android/canonical/app/src/androidTest/java/com/google/samples/quickstart/canonical/DatabaseInteractionTest.kt

```diff
@@ -0,0 +1,83 @@
+package com.google.samples.quickstart.canonical
+
+import androidx.test.espresso.Espresso.onData
+import androidx.test.espresso.Espresso.onView
+import androidx.test.espresso.action.ViewActions.click
+import androidx.test.espresso.assertion.ViewAssertions
+import androidx.test.espresso.assertion.ViewAssertions.matches
+import androidx.test.espresso.matcher.RootMatchers.isDialog
+import androidx.test.espresso.matcher.ViewMatchers
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import androidx.test.platform.app.InstrumentationRegistry
+import androidx.test.rule.ActivityTestRule
+import androidx.test.uiautomator.UiDevice
+import androidx.test.uiautomator.UiSelector
+import org.hamcrest.Matchers
+import org.junit.After
+import org.junit.Assert.*
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+import org.junit.runner.RunWith
+
+@RunWith(AndroidJUnit4::class)
+class DatabaseInteractionTest {
+    @get:Rule
+    var activityRule: ActivityTestRule<MainActivity>
+            = ActivityTestRule(MainActivity::class.java)
+
+    lateinit var device : UiDevice
+
+    @Before
+    fun setup() {
+        device = UiDevice.getInstance(InstrumentationRegistry.getInstrumentation())
+        onView(ViewMatchers.withId(R.id.sign_in_button))
+            .perform(click())
+        val googleSignInDialog = device.findObject(UiSelector().text("yl4324@columbia.edu"))
```

Comment:
You should probably replace with an email address that isn't your own. e.g. example@gmail.com.

Ideally when running the test, a test account will be setup and signed in on the device.